cheribsd/Jenkinsfile

```diff
@@ -11,6 +11,13 @@ find .
 for f in cheribsd-sysroot.tar.xz cheribsd-${suffix}.img.xz kernel.xz; do
     mv -fv artifacts-${suffix}/\$f .
 done
+"""
+    }
+    stage("decompress kernel and qemu-img convert disk image") {
+        sh """
+unxz kernel.xz
+unxz cheribsd-${suffix}.img.xz
+/cheri-sdk/bin/qemu-img convert -o compression_type=zstd -O qcow2 -c cheribsd-${suffix}.{img,qcow2}
```

Comment:
Oh, hm, technically this is a bit dodgy, the QEMU build is for whatever version of Ubuntu is used in the container, which may not match the host. I don't know how much qemu-img ends up pulling in and thus whether it's linking against any libraries that see ABI churn (or have seen rare churn, such as ncurses that bumped soname between 18.04 and 20.04).

Reply:
Sadly true.  I just checked and the Ubuntu `20.04` `qemu-img` is version 4.2 which predates `zstd` compression.  We *could* do this by spawning a docker process of the appropriate flavour?  Perhaps something like this, if I read https://www.jenkins.io/doc/book/pipeline/docker/ correctly?

```
stage("qemu-img convert disk image") {
  agent {
    docker {
      image 'ctsrd/cheri-qemu:latest'
      args '-v $PWD:/mnt -u $UID:1'
    }
  }
  steps {
    sh '/cheri-sdk/bin/qemu-img convert -o compression_type=zstd -O qcow2 -c /mnt/cheribsd-${suffix}.{img,qcow2}
  }
}
```

Bets on whether or not that would work?

Reply:
I'd probably just shove it in the Dockerfile and take advantage of multi-stage builds